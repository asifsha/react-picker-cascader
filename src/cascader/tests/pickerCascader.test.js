import React from "react";
import PickerCascader from "../index";
import { expect, it } from "@jest/globals";

import renderer from 'react-test-renderer'


const pcData = [
  {
    key: "1",
    text: "Australia",
    children: [
      {
        key: "2",
        text: "New South Wales",
        children: [
          { key: "3", text: "Sydney" },
          { key: "4", text: "Wollongong" }
        ]
      },
      {
        key: "5",
        text: "Victoria",
        children: [
          { key: "6", text: "Melbourne" },
          { key: "7", text: "Geelong" }
        ]
      }
    ]
  },
  {
    key: "10",
    text: "Canada",
    children: [
      {
        key: "11",
        text: "Alberta",
        children: [
          { key: "12", text: "Calgary" },
          { key: "13", text: "Brooks" }
        ]
      },
      {
        key: "14",
        text: "British Columbia",
        children: [
          { key: "15", text: "Vancouver" },
          { key: "16", text: "Vernon" }
        ]
      }
    ]
  },
  {
    key: "20",
    text: "United States",
    children: [
      {
        key: "21",
        text: "New York",
        children: [
          { key: "22", text: "Albany" },
          { key: "23", text: "Norwich" }
        ]
      },
      {
        key: "24",
        text: "Pennsylvania",
        children: [
          { key: "25", text: "Farrell" },
          { key: "26", text: "Parker" }
        ]
      }
    ]
  }
];

/*global done : true*/
it('renders correctly with data', () => {
  const tree = renderer.create(<PickerCascader data={pcData} onValueSelected={() => done()} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly without data', () => {
  const tree = renderer.create(<PickerCascader data={[]} onValueSelected={() => done()} />).toJSON();
  expect(tree).toMatchSnapshot();
});
