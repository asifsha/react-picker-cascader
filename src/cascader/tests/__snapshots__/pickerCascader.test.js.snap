// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with data 1`] = `
<div
  style={
    Object {
      "width": "20rem",
    }
  }
>
  <div
    className="pc-text-panel"
    onClick={[Function]}
    style={
      Object {
        "color": "grey",
      }
    }
  >
    <span
      className="pc-dropdown-icon"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 320 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
        />
      </svg>
    </span>
  </div>
  <div
    className="pc-list-wrapper"
  />
</div>
`;

exports[`renders correctly without data 1`] = `
<div
  style={
    Object {
      "width": "20rem",
    }
  }
>
  <div
    className="pc-text-panel"
    onClick={[Function]}
    style={
      Object {
        "color": "grey",
      }
    }
  >
    <span
      className="pc-dropdown-icon"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 320 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
        />
      </svg>
    </span>
  </div>
  <div
    className="pc-list-wrapper"
  />
</div>
`;
